File: atcoder/abc133B_good_distance.nim
# uninhm
# https://atcoder.jp/contests/abc133/tasks/abc133_b
# brute force

import strutils, sequtils
from math import `^`, sqrt

func getDistance(a, b: seq[int]): float =
  for i in 0 ..< a.len:
    result += float(abs(a[i] - b[i]) ^ 2)
  result = sqrt(result)

var n, d: int
(n, d) = stdin.readLine.split.map(parseInt)

var
  x = newSeq[seq[int]](n)
  ans = 0
  distance: float
  

for i in 0 ..< n:
  x[i] = stdin.readLine.split.map(parseInt)

for i in 0 ..< n:
  for j in i+1 ..< n:
    distance = x[i].getDistance(x[j])
    if distance == distance.int.float:
      inc ans

ans.echo

File: atcoder/abc133B_good_distance.ml
(* vicfred
 * https://atcoder.jp/contests/abc133/tasks/abc133_b
 * math, implementation
 * *)
let ans, n, xss = Scanf.(scanf "%d %d" @@ fun n d -> ref 0, n, Array.(init n @@ fun _ -> init d @@ fun _ -> scanf " %d" ( + ) 0))
let rec f i n = if i * i >= n then i * i  = n else f (i + 1) n
let g n = n * n
let main = Array.(iteri (fun i xs -> for j = i +  1 to n - 1 do if f 1 @@ fold_left ( + ) 0 @@ mapi (fun k x -> g (x - xss.(j).(k))) xs then incr ans done) xss); Printf.printf "%d\n" !ans

